nixos/modules/misc/locate.nix

```diff
@@ -244,10 +247,10 @@ in {
           ''
           else ''
             exec ${cfg.locate}/bin/updatedb \
-              ${optionalString (cfg.localuser != null && ! isMLocate) "--localuser=${cfg.localuser}"} \
+              ${optionalString (cfg.localuser != null && ! isMorPLocate) "--localuser=${cfg.localuser}"} \
               --output=${toString cfg.output} ${concatStringsSep " " cfg.extraFlags}
           '';
-        environment = optionalAttrs (!isMLocate) {
+        environment = optionalAttrs (!isMorPLocate) {
           PRUNEFS = concatStringsSep " " cfg.pruneFS;
           PRUNEPATHS = concatStringsSep " " cfg.prunePaths;
           PRUNENAMES = concatStringsSep " " cfg.pruneNames;
```

Comment:
Can you also carry over https://github.com/SuperSandro2000/nixpkgs/commit/0f7c24a12133e191903c14cdd4ce9a982e692c67#diff-7b0b4fcaa808c3b1572ab48be3a2f5fb12a79f0e1930c124fe2b7b0136df8996R291 ? If you want, you can also add yourself.

Reply:
Done. I have also fixed all the cases where I did not properly merge the `mkIf` expressions.